crates/get-size2/src/lib.rs

```diff
@@ -518,3 +518,10 @@ where
         self.iter().map(GetSize::get_size).sum()
     }
 }
+
+#[cfg(feature = "url")]
+impl GetSize for url::Url {
+    fn get_heap_size(&self) -> usize {
+        self.as_str().len()
```

Comment:
`Url` actually contains a `String`, but there is no way to obtain it without doing a `clone` and `into`, which seems a bit too expensive to me. Instead I opted to use the `str` length instead. How do you want to approach this?